File: src/fixengine/messages/fix42/hotspotfx/ExecutionReport.java
/*
 * Copyright 2010 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package fixengine.messages.fix42.hotspotfx;

import fixengine.messages.MessageHeader;
import fixengine.messages.Required;
import fixengine.tags.fix42.Account;
import fixengine.tags.fix42.AvgPx;
import fixengine.tags.fix42.ClOrdID;
import fixengine.tags.fix42.ContraBroker;
import fixengine.tags.fix42.CumQty;
import fixengine.tags.fix42.Currency;
import fixengine.tags.fix42.ExecBroker;
import fixengine.tags.fix42.ExecID;
import fixengine.tags.fix42.ExecInst;
import fixengine.tags.fix42.ExecTransType;
import fixengine.tags.fix42.ExecType;
import fixengine.tags.fix42.ExpireTime;
import fixengine.tags.fix42.FutSettDate;
import fixengine.tags.fix42.LastPx;
import fixengine.tags.fix42.LastShares;
import fixengine.tags.fix42.LeavesQty;
import fixengine.tags.fix42.MaxShow;
import fixengine.tags.fix42.MinQty;
import fixengine.tags.fix42.NoContraBrokers;
import fixengine.tags.fix42.OrdStatus;
import fixengine.tags.fix42.OrderID;
import fixengine.tags.fix42.OrderQty2;
import fixengine.tags.fix42.OrderQty;
import fixengine.tags.fix42.OrigClOrdID;
import fixengine.tags.fix42.Price;
import fixengine.tags.fix42.SecurityType;
import fixengine.tags.fix42.SettlCurrAmt;
import fixengine.tags.fix42.SettlCurrency;
import fixengine.tags.fix42.Side;
import fixengine.tags.fix42.Symbol;
import fixengine.tags.fix42.Text;
import fixengine.tags.fix42.TimeInForce;
import fixengine.tags.fix42.TradeDate;
import fixengine.tags.fix42.TransactTime;
import fixengine.tags.fix44.TradeLinkID;

public class ExecutionReport extends fixengine.messages.fix42.ExecutionReport {
    public ExecutionReport(MessageHeader header) {
        super(header);
    }

    @Override protected void fields() {
        field(Account.Tag(), Required.NO);
        field(ClOrdID.Tag());
        field(ExecID.Tag());
        field(ExecInst.Tag(), Required.NO);
        field(ExecTransType.Tag());
        field(OrderID.Tag());
        field(OrdStatus.Tag());
        field(OrigClOrdID.Tag(), Required.NO);
        field(OrderQty.Tag(), Required.NO);
        field(Price.Tag(), Required.NO);
        field(TimeInForce.Tag(), Required.NO);
        field(Currency.Tag(), Required.NO);
        field(Side.Tag());
        field(Symbol.Tag());
        field(ExecBroker.Tag(), Required.NO);
        field(LeavesQty.Tag());
        field(CumQty.Tag());
        field(MinQty.Tag(), Required.NO);
        field(OrderQty2.Tag(), Required.NO);
        field(FutSettDate.Tag(), Required.NO);
        field(TradeDate.Tag(), Required.NO);
        field(SettlCurrAmt.Tag(), Required.NO);
        field(SettlCurrency.Tag(), Required.NO);
        field(SecurityType.Tag());
        field(MaxShow.Tag(), Required.NO);
        field(ExecType.Tag());
        field(AvgPx.Tag());
        field(LastShares.Tag(), Required.NO);
        field(LastPx.Tag(), Required.NO);
        field(NoContraBrokers.Tag(), Required.NO);
        field(ContraBroker.Tag(), Required.NO);
        field(Text.Tag(), Required.NO);
        field(TransactTime.Tag());
        field(ExpireTime.Tag(), Required.NO);
        field(TradeLinkID.Tag(), Required.NO);
    }
}

File: src/main/scala/xtch/turquoise/templates/ExecutionReport.scala
/*
 * Copyright 2010 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package xtch.turquoise.templates

import xtch.turquoise.Fields
import xtch.turquoise.MessageType

/**
 * Template for Execution Report message as specified in section 7.4.5 of
 * [2].
 */
object ExecutionReport extends AbstractTemplate(MessageType.EXECUTION_REPORT) {
  add(MessageHeader.TEMPLATE);
  add(Fields.APP_ID);
  add(Fields.SEQUENCE_NO);
  add(Fields.EXECUTION_ID);
  add(Fields.CLIENT_ORDER_ID);
  add(Fields.ORDER_ID);
  add(Fields.EXEC_TYPE);
  add(Fields.EXECUTION_REPORT_REF_ID);
  add(Fields.ORDER_STATUS);
  add(Fields.ORDER_REJECT_CODE);
  add(Fields.EXECUTED_PRICE);
  add(Fields.EXECUTED_QTY);
  add(Fields.LEAVES_QTY);
  add(Fields.RESERVED_FIELD_1);
  add(Fields.DISPLAY_QTY);
  add(Fields.COMMON_SYMBOL);
  add(Fields.SIDE);
  add(Fields.SECONDARY_ORDER_ID);
  add(Fields.COUNTERPARTY);
  add(Fields.TRADE_LIQUIDITY_INDICATOR);
  add(Fields.TRADE_MATCH_ID);
  add(Fields.TRANSACT_TIME);
  add(Fields.TARGET_BOOK);
  add(Fields.TYPE_OF_TRADE);
  add(Fields.RESERVED_FIELD_1);
}
